server/src/main/java/org/elasticsearch/cluster/DiffableUtils.java

```diff
@@ -402,6 +402,25 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    private static <K, T> Map.Entry<K, T> mapEntry(K key, T newValue) {
```

Comment:
Why the special map entry? These will be recreated by any map impls right?

Reply:
Yea, we really didn't need `Map.Entry` for the sites that we use it to begin with, `Tuple` would've been just fine but I didn't want to mix more refactoring than necessary into this PR so I went with this since weirdly enough I couldn't find a built-in JDK way of creating a `Map.Entry` that allows me a `null` value ...